### PR TITLE
Ignore fees per channel

### PR DESCRIPTION
**Pull request by rustyrussell**
We always had a global `ignore-fee-limits`: this adds an equivalent on a per-channel basis.  This is a workaround, of course: letting peers set arbitrary fees is fraught, but there are cases where this can help.  For example, there seems to be a problem where LND nodes insist on retransmitting the same fee, even if it knows it's now wrong (e.g. neutrino, or bitcoind restart).  We will keep closing the connection on them.  

With this, you could temporarily `setchannel id=<channel> ignorefeelimits=true` and it will hopefully make progress.  You can also use it to "trust" peers, and not get upset if there are fee disagreements.

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
If you have time to rebase it, I will give a shot for the review today :)

**Comment by rustyrussell** [comment]
> If you have time to rebase it, I will give a shot for the review today :)

Done, thanks!

**Comment by wtogami** [comment]
https://fedorapeople.org/~wtogami/a/2023/0001-DO-NOT-COMMIT-Temporary-workaround-for-LND-update_fe.patch
Ugly workaround for v23.05.2

This `setchannel` approach of runtime switchable ignore fees is far superior to an ugly hack that requires restarting CLN.

Would it be possible to add a print to let you know every time an ignore fee event happened?

What I like about my ugly hack is the print tells you exactly what is happening. It is diagnostically useful to know because it should happen JUST ONCE per channel to recover from LND's current bug after their fee estimator was broken **in the past**. If it happens more than once then the remote peer's fee estimator is **currently** broken. You want to know that so there is a chance to talk to the sysadmin to be able to fix it. It appears this PR currently would be silent when it ignores which is less informative?

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -748,6 +748,9 @@ static void json_add_channel(struct lightningd *ld,
 		json_add_node_id(response, "peer_id", &peer->id);
 		json_add_bool(response, "peer_connected", peer->connected == PEER_CONNECTED);
 		json_add_channel_type(response, "channel_type", channel->type);
+		if (channel->ignore_fee_limits) {
+			json_add_bool(response, "ignore_fee_limits", channel->ignore_fee_limits);
+		}
```
Comment (vincenzopalazzo) [inline lightningd/peer_control.c:753]: Did it make sense to make this optional? We add this when the `ingore_fee_limits` only when it is true, why do we not add this also when it is false?

removing the if directly.
```c
json_add_bool(response, "ignore_fee_limits", channel->ignore_fee_limits);
```

usually, as a user, I use the JSON response to understand what I can do, and then I jump inside the docs to see how I can enable it

File: lightningd/peer_control.c [hunk 2]
```diff
@@ -2789,6 +2798,7 @@ static struct command_result *json_setchannel(struct command *cmd,
 		   p_opt("htlcmin", param_msat, &htlc_min),
 		   p_opt("htlcmax", param_msat, &htlc_max),
 		   p_opt_def("enforcedelay", param_number, &delaysecs, 600),
+		   p_opt("ignorefeelimits", param_bool, &ignore_fee_limits),
```
Comment (vincenzopalazzo) [inline lightningd/peer_control.c:2801]: Why do we not use the `ignore_fee_limits` as keyword? I feel that using this one-word style, we will make the autogen code a little bit ugly?

**Review by vincenzopalazzo — APPROVED** [review]
The critical logic LGMT but I had some comments from the API point of view.

I was wondering if the following CI failure is related?

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/test_opening.py::test_anchor_min_emergency - ValueError: 
Node errors:
 - lightningd-1: had bad gossip messages
Global errors:
======= 59 passed, 3 skipped, 4 warnings, 1 error in 2441.54s (0:40:41) ========
```

File: lightningd/peer_control.c (hunk 2 above)
Comment (rustyrussell) [reply lightningd/peer_control.c:2801]: It's consistent though :(

File: lightningd/peer_control.c (hunk 1 above)
Comment (rustyrussell) [reply lightningd/peer_control.c:753]: We could do either.  I don't want to promote this, do I hide it unless it's disabled.

In the long run, this should become unnecessary!

File: lightningd/peer_control.c (hunk 2 above)
Comment (vincenzopalazzo) [reply lightningd/peer_control.c:2801]: yeah I know, now there is the was of pre-code gen and post code gen ahha

File: lightningd/peer_control.c (hunk 1 above)
Comment (vincenzopalazzo) [reply lightningd/peer_control.c:753]: I see I see

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6398/commits/7ec9faaa1ba06600f452e037bd84fcd4cc3c88a7

**Comment by rustyrussell** [comment]
> https://fedorapeople.org/~wtogami/a/2023/0001-DO-NOT-COMMIT-Temporary-workaround-for-LND-update_fe.patch Temporary workaround for v23.05.2
> 
> Your `setchannel` approach of runtime switchable ignore fees is far superior to temporary hack that requires restarting CLN.
> 
> Would it be possible to add a print to let you know every time an ignore fee event happened?
> 
> What I like about my ugly hack is the print tells you exactly what is happening. It is diagnostically useful to know because it should happen JUST ONCE per channel to recover from LND's bug after their fee estimator was broken **in the past**. If it happens more than once the remote peer's fee estimator is **currently** broken. You want to know that so there is a chance to talk to the sysadmin to be able to fix it. It appears this PR currently would be silent when it ignores which is less informative?

This may be possible, but the current PR works by telling channeld an infinite allowable range. I think we can do logging in lightningd however, if the actual feerate used is weird.

I will do this as a separate PR.

**Pull request merged**